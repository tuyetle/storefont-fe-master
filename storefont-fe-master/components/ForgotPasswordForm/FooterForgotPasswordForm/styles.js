import { css } from 'glamor';
import variables from '~/styles/variables';

export default {
  footer: css({
    paddingBottom: '2em',
  }),
  backIcon: css({
    verticalAlign: 'middle',
  }),
  loginButton: css({
    background: 'none',
    color: variables.$locationTextColor,
    border: 'none',
    textDecoration: 'none',
    textTransform: 'uppercase',
    padding: '0 0.2em',
    fontWeight: 'bold',
    cursor: 'pointer',
    outline: '0',
    fontSize: '0.813em',
    '@media(min-width: 768px)': {
      color: variables.$colorWelcome,
      fontSize: '0.938em',
    },
  }),
};
