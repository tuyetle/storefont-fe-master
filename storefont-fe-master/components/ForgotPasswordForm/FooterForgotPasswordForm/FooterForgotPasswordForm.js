import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import { noop } from 'lodash';
import BackIcon from '~/static/assets/img/ic-arrow-back-black.svg';
import styles from './styles';

class FooterForgotPasswordForm extends PureComponent {
  showLoginHandler = (e) => {
    e.preventDefault();
    this.props.showModal({ currentModal: 'login' });
  }
  render() {
    return (
      <div className={`text-center ${styles.footer}`}>
        <BackIcon className={styles.backIcon} />
        <button
          className={styles.loginButton}
          onClick={this.showLoginHandler}
        >
          {this.props.t('login')}
        </button>
      </div>
    );
  }
}

FooterForgotPasswordForm.propTypes = {
  showModal: PropTypes.func,
  t: PropTypes.func,
};
FooterForgotPasswordForm.defaultProps = {
  showModal: noop,
  t: noop,
};
export default translate(['auth'])(FooterForgotPasswordForm);

