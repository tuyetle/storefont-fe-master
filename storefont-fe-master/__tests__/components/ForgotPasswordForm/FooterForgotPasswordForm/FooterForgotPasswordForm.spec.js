import React from 'react';
import FooterForgotPasswordForm from '~/components/ForgotPasswordForm/FooterForgotPasswordForm/FooterForgotPasswordForm';
import renderer from 'react-test-renderer';

describe('(Component) FooterForgotPasswordForm', () => {
  let component;
  let props;
  let spyCallback;

  beforeEach(() => {
    props = {
      t(value) { return value; },
      showModal() {},
    };
    spyCallback = jest.spyOn(props, 'showModal');
    component = mount(<FooterForgotPasswordForm {...props} />);
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should call callback when select', () => {
    const btn = component.find('button');
    btn.simulate('click');
    expect(spyCallback).toHaveBeenCalled();
  });
});
