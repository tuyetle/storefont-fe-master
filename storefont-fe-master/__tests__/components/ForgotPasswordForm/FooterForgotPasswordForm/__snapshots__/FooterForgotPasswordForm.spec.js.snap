// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) FooterForgotPasswordForm should render conrectly 1`] = `
<div
  className="text-center css-zo3d3e"
>
  <svg
    className={
      Object {
        "data-css-1lz3n2r": "",
      }
    }
    height="22"
    width="36"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.993 17.921a.964.964 0 0 0 1.34 0 .887.887 0 0 0 0-1.277L3.538 10.1H25.96c.523-.001.938-.401.938-.904a.925.925 0 0 0-.938-.916H3.538l6.795-6.531a.903.903 0 0 0 0-1.291.964.964 0 0 0-1.34 0L.59 8.55a.867.867 0 0 0 0 1.278l8.403 8.093z"
      fill="#606060"
      fillRule="evenodd"
    />
  </svg>
  <button
    className={
      Object {
        "data-css-kw9tnt": "",
      }
    }
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
